src/Chapter2.hs

```diff
@@ -335,8 +335,11 @@ from it!
 
 ghci> :l src/Chapter2.hs
 -}
+
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList x y list
+  | x >= 0 = (take (y - x + 1) . drop x) list
```

Comment:
Your implementation is correct when the right bound is lower than the left. But I think you can implement it more efficiently by adding a separate case to your check :slightly_smiling_face:

Reply:
I think I got it, making the checks first makes the code more readable